src/monodroid/jni/monodroid-networkinfo.c

```diff
@@ -0,0 +1,89 @@
+/*
+ * monodroid-networkinfo.c
+ *
+ * Authors:
+ *      Marek Habersack <grendel@twistedcode.net>
+ *
+ * Copyright (C) 2016 Microsoft (http://microsoft.com)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining
+ * a copy of this software and associated documentation files (the
+ * "Software"), to deal in the Software without restriction, including
+ * without limitation the rights to use, copy, modify, merge, publish,
+ * distribute, sublicense, and/or sell copies of the Software, and to
+ * permit persons to whom the Software is furnished to do so, subject to
+ * the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+ * LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+ * WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include <string.h>
+
+#include "monodroid.h"
+#include "logger.h"
+#include "monodroid-glue.h"
+#include "util.h"
+#include "dylib-mono.h"
+
+static mono_bool
+_monodroid_get_network_interface_state (const char *ifname, mono_bool *is_up, mono_bool *supports_multicast)
+{
+	if (!ifname || strlen (ifname) == 0 || (!is_up && !supports_multicast))
+		return FALSE;
+
+	mono_bool ret = TRUE;
+
+	if (is_up)
+		*is_up = FALSE;
+	if (supports_multicast)
+		*supports_multicast = FALSE;
+
+	JNIEnv *env = get_jnienv ();
+	jclass NetworkInterface_class = (*env)->FindClass (env, "java/net/NetworkInterface");
```

Comment:
It could, but is there a point? I guess it depends on how much of a performance hit it is. I chose to keep the code together for clarity, but if you think we should cache the class reference, then sure :)